lib/workers/evm_server.rb

```diff
@@ -18,17 +21,279 @@ def start
 
     PidFile.create(MiqServer.pidfile)
     set_process_title
-    MiqServer.start
+    validate_database
+    EvmDatabase.seed_primordial
+    check_migrations_up_to_date
+
+    start_servers
+    monitor_servers
+  rescue Interrupt => e
+    _log.info("Received #{e.message} signal, killing server")
+    kill_servers
+    exit 1
+  rescue SignalException => e
+    _log.info("Received #{e.message} signal, shutting down server")
+    stop_servers
+    exit 0
   end
 
-  ##
-  # Sets the server process' name if it is possible.
-  #
-  def set_process_title
-    Process.setproctitle(SERVER_PROCESS_TITLE) if Process.respond_to?(:setproctitle)
+  def start_servers
+    refresh_servers_to_monitor
+    for_each_server { start_server }
```

Comment:
I'm amazed if this `for_each_server` loop works for all of the things you're doing in blocks.  💯 points if you get it to work!